src/javascript/app/pages/user/account/settings/personal_details.js

```diff
@@ -54,6 +59,70 @@ const PersonalDetails = (() => {
         $('#missing_details_notice').setVisibility(!!has_missing_field);
     };
 
+    const populateChangeableFields = () => {
+        if (!has_changeable_fields) return;
+
+        const landing_companies = State.getResponse('landing_company');
+        const changeable        = landing_companies.financial_company.changeable_fields;
+        if (changeable && changeable.only_before_auth) {
+            changeable_fields = changeable.only_before_auth;
+        }
+    };
+
+    /**
+     * Remove labels and static fields and replace them with input when fields are changeable.
+     *
+     * @param {get_settings} to prepopulate some of the values.
+     */
+    const displayChangeableFields = (get_settings) => {
+        if (!has_changeable_fields) return;
+        changeable_fields.forEach(field => {
+            CommonFunctions.getElementById(`row_${field}`).setVisibility(1);
+        });
+
+        if (changeable_fields.includes('first_name') || changeable_fields.includes('last_name') || changeable_fields.includes('salutation')) {
+            CommonFunctions.getElementById('row_name').setVisibility(0);
+            CommonFunctions.getElementById('first_name').setVisibility(1);
+            CommonFunctions.getElementById('last_name').setVisibility(1);
+            CommonFunctions.getElementById('salutation').setVisibility(1);
+        }
+
+        if (changeable_fields.includes('date_of_birth')) {
+            const $input_el = $('#date_of_birth');
+            $input_el.setVisibility(1);
+            $input_el
+                .attr('data-value', toISOFormat(moment().subtract(18, 'years')))
+                .change(function() {
+                    return CommonFunctions.dateValueChanged(this, 'date');
+                });
+
+            DatePicker.init({
+                selector: '#date_of_birth',
+                minDate : -100 * 365,
+                maxDate : (
+                    -18 * 365
+                ) - 5,
```

Comment:
can you explain this?

Reply:
I don't get it. Do you want me to add a comment in the code? or just explain it here?

Reply:
comment in the code :)